themis_rpc.go

```diff
@@ -57,30 +48,28 @@ func (t *themisClientImpl) call(methodName string, tbl, row []byte, req pb.Messa
 	return nil
 }
 
-func (t *themisClientImpl) checkAndSetLockIsExpired(lock ThemisLock) (bool, error) {
-	b, err := t.isLockExpired(lock.getColumn().Table, lock.getColumn().Row, lock.getTimestamp())
-	if err != nil {
-		return false, errors.Trace(err)
-	}
-	lock.setExpired(b)
-	return b, nil
+func (rpc *themisRPC) checkAndSetLockIsExpired(lock Lock) (bool, error) {
+	ts := lock.Timestamp() >> 18
+	expired := rpc.oracle.IsExpired(ts, rpc.conf.TTLInMs)
```

Comment:
I think it is better to pass lock.Timestamp() directly and do byte shift inside oracle. Since we have defined constant for 18 inside package oracles.